src/aics-image-viewer/shared/utils/playControls.ts

```diff
@@ -0,0 +1,80 @@
+import { AxisName } from "../types";
+
+type PlayAxisName = AxisName | "t";
+
+const PLAY_STEP_INTERVAL = 125;
+
+export default class PlayControls {
+  playingAxis: PlayAxisName | null = null;
+  playWaitingForLoad = false;
+  playHolding = false;
+  playTimeoutId = 0;
+
+  public getVolumeIsLoaded?: () => boolean;
+  public stepAxis?: (axis: PlayAxisName) => void;
+  public onPlayingAxisChanged?: (axis: PlayAxisName | null) => void;
+
+  private setPlayingAxis(axis: PlayAxisName | null): void {
+    this.playingAxis = axis;
+    this.onPlayingAxisChanged?.(axis);
+  }
+
+  private playStep(): void {
+    if (!this.playingAxis || this.playHolding || !this.stepAxis) {
+      return;
+    }
+    // If the volume is not loaded, wait for it to load before continuing
+    if (!this.getVolumeIsLoaded?.()) {
+      this.playWaitingForLoad = true;
+      return;
+    }
+
+    this.stepAxis(this.playingAxis);
+    this.playTimeoutId = window.setTimeout(this.playStep.bind(this), PLAY_STEP_INTERVAL);
+  }
+
+  /** Call whenever new data is loaded to resume playback if it was paused for data loading. */
+  onImageLoaded(): void {
+    if (this.playWaitingForLoad) {
+      this.playWaitingForLoad = false;
+      this.playStep();
+    }
+  }
+
+  /**
+   * Pause playback on the currently playing axis.
+   * `willResume` marks this as a temporary suspension, e.g. while the user is scrubbing along the playing axis.
+   */
+  pause(willResume = false): void {
```

Comment:
I like the language of `pause` and `stop`... maybe easier to read than adding the boolean flag here

Reply:
I kinda wonder if renaming `willResume` would make this clearer, but the comment seems clear enough too.  Could even call it `isTemporary` (or the negation)